src/mender-update/daemon/states.cpp

```diff
@@ -953,7 +953,13 @@ ExitState::ExitState(events::EventLoop &event_loop) :
 }
 
 void ExitState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
+#ifndef NDEBUG
+	if (--iterations_left_ <= 0) {
+		event_loop_.Stop();
+	}
```

Comment:
That's a very good point! It should not continue, and it may indicate a bug elsewhere. I'll look into it.

Reply:
It was my own fault. [Here](https://github.com/mendersoftware/mender/commit/22f1b7932d5dafac9988d7d49904cd442efe1fdc#diff-8b9f2c6e24ffc9c97c30370e3129098c5eb1f6fc06ba4e1008d344d077358f4bR465) I mistyped the event and made it react to `DeploymentEnded`, instead of `Success`. Because `DeploymentEnded` was also in the queue because of the `deployment_tracking_` state machine, it worked by accident, since `Success` was then not consumed and left in the queue for the next state to consume instead.

Fixed by changing the transition to react to the correct event, and posting the `Success` event, as we should.